sqlmesh/core/model/definition.py

```diff
@@ -1053,6 +1059,7 @@ def update_schema(
         )
         self._columns_to_types = None
         self._query_renderer._optimized_cache = {}
+        self._query_renderer.update_schema(self.mapping_schema)
```

Comment:
Was this a bug?

Reply:
Should 
```
self._query_renderer._optimized_cache = {}
```
happen inside the renderer's `update_schema` then?

Reply:
no, the code used to generate mapping schema every time you call render / macro evaluator etc. so it was sharing the mapping_schema dict and renormalizing the mapping schema. now i only do it once or when the schema is updated. it's cleaner this way anyways.

Reply:
ah, right. I didn't like the fact that we had to rely on the mutable nature of that dict. Good 👍